ui/ducks/institutional/institutional.js

```diff
@@ -0,0 +1,114 @@
+import { createSelector } from 'reselect';
+import { createSlice } from '@reduxjs/toolkit';
+import { captureException } from '@sentry/browser';
+import { mmiActionsFactory } from '../../store/institutional/institution-background';
+
+const createComplianceActionType = (action) =>
+  `metamask/institutional-features/compliance/${action}`;
+
+const SET_HISTORICAL_REPORTS = createComplianceActionType(
+  'SET_HISTORICAL_REPORTS',
+);
+
+const name = 'institutionalFeatures';
+
+const initialState = {
+  historicalReports: {},
+  complianceProjectId: '',
+  complianceClientId: '',
+  reportsInProgress: {},
+};
+
+const slice = createSlice({
+  name: 'reports',
+  initialState,
+  reducers: {
+    setHistoricalReports(state, action) {
+      state.historicalReports[action.payload.address] = [
+        ...action.payload.reports,
+      ];
+    },
+  },
+});
+
+const { actions, reducer } = slice;
+
+export default reducer;
+
+export const getComplianceProjectId = (state) =>
+  state.metamask[name].complianceProjectId;
+export const getComplianceClientId = (state) =>
+  state.metamask[name].complianceClientId;
+export const getComplianceTenantSubdomain = (state) =>
+  state.metamask[name].complianceTenantSubdomain;
+export const getComplianceHistoricalReports = (state) =>
+  state.metamask[name].historicalReports;
+export const getComplianceReportsInProgress = (state) =>
+  state.metamask[name].reportsInProgress;
+export const getInstitutionalConnectRequests = (state) =>
+  state.metamask[name].connectRequests;
+export const complianceActivated = (state) =>
+  Boolean(state.metamask[name].complianceProjectId);
+
+export const getComplianceHistoricalReportsByAddress = (address) =>
+  createSelector(getComplianceHistoricalReports, (reports) =>
+    reports ? reports[address] : [undefined],
```

Comment:
why `[undefined]` instead of an empty array? usually, we use `null` to assign a no-value value 🤔

Reply:
This is fixed in this PR https://github.com/MetaMask/metamask-extension/pull/18578